File: tsp-solver-frontend/src/app/model/request/tsp/SimulatedAnnealingRequest.ts
export class SimulatedAnnealingRequest {
  costMatrix: number[][]
  maxTemperature: number;
  minTemperature: number;
  coolingRate: number;
  iterations: number
  maxCoolingTemperatureNoImprovement: number

  constructor(costMatrix: number[][], maxTemperature: number, minTemperature: number,
              coolingRate: number, iterations: number, maxCoolingTemperatureNoImprovement: number) {
    this.costMatrix = costMatrix;
    this.maxTemperature = maxTemperature;
    this.minTemperature = minTemperature;
    this.coolingRate = coolingRate;
    this.iterations = iterations;
    this.maxCoolingTemperatureNoImprovement = maxCoolingTemperatureNoImprovement;
  }

}

File: tsp-solver-backend/src/main/java/pl/edu/pbs/request/tsp/SimulatedAnnealingRequest.java
package pl.edu.pbs.request.tsp;

import lombok.Data;

@Data
public class SimulatedAnnealingRequest {

    private double[][] costMatrix;
    private double maxTemperature;
    private double minTemperature;
    private double coolingRate;
    private int iterations;
    private int maxCoolingTemperatureNoImprovement;

}